common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -260,9 +262,11 @@ def _get_enabled_provider_by_name(provider_name):
     return enabled_provider
 
 
-def _get_url(view_name, backend_name, auth_entry=None, redirect_url=None, enroll_course_id=None, email_opt_in=None):
+def _get_url(view_name, backend_name, auth_entry=None, redirect_url=None,
+             enroll_course_id=None, email_opt_in=None, extra_params=None,
+             **kwargs):
```

Comment:
The parameter list of this function pollutes the namespace of the `kwargs` parameter: everything occurring as a parameter name can't be in `kwargs`.  This is the reason why Django's `reverse()` uses the slightly awkward `kwargs=dict(...)` syntax instead of accepting keyword arguments directly, and I suggest that you do the same here.